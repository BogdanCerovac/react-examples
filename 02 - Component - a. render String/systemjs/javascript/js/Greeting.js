/*! Mozilla Public License Version 2.0 !*/
/*! Copyright Â© 2018 Rick Beerendonk   !*/

function Greeting() {
  return 'Hello World!';
}
export default Greeting;
