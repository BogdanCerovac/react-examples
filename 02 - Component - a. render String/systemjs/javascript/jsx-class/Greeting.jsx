/*! Mozilla Public License Version 2.0 !*/
/*! Copyright Â© 2018 Rick Beerendonk   !*/

import React from 'react';

class Greeting extends React.Component {
  render() {
    return 'Hello World!';
  }
}

export default Greeting;
