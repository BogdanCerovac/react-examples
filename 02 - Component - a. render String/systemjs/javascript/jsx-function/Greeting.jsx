/*! Mozilla Public License Version 2.0 !*/
/*! Copyright Â© 2018 Rick Beerendonk   !*/

const Greeting = () => 'Hello World!';

export default Greeting;
