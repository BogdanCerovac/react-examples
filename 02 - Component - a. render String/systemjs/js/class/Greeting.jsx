/*! European Union Public License version 1.2 !*/
/*! Copyright Â© 2018 Rick Beerendonk          !*/

import React from 'react';

class Greeting extends React.Component {
  render() {
    return 'Hello World!';
  }
}

export default Greeting;
