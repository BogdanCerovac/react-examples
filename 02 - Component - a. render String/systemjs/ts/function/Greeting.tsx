/*! European Union Public License version 1.2 !*/
/*! Copyright Â© 2018 Rick Beerendonk          !*/

function Greeting(): string {
  return 'Hello World!';
}

export default Greeting;
