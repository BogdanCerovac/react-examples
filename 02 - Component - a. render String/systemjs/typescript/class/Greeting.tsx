/*! Mozilla Public License Version 2.0 !*/
/*! Copyright Â© 2018 Rick Beerendonk   !*/

import React from 'react';

class Greeting extends React.Component<undefined, undefined> {
  render(): string {
    return 'Hello World!';
  }
}

export default Greeting;
