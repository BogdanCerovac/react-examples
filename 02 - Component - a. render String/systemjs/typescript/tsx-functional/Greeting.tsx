/*! Mozilla Public License Version 2.0 !*/
/*! Copyright Â© 2018 Rick Beerendonk   !*/

import React from 'react';

const Greeting = (): string => 'Hello World!';

export default Greeting;
