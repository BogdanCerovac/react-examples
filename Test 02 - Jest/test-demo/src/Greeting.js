import React from 'react';

/* eslint react/prop-types: "off" */

class Greeting extends React.Component {
  constructor(props) {
    super(props);

    if (!props.name) {
      throw new Error('No name provided');
    }
  }
  render() {
    return <h1>Hello {this.props.name}!</h1>;
  }
}

export default Greeting;
