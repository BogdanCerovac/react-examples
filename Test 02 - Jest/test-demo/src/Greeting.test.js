import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Greeting from './Greeting';

/// Enzyme ///

it('renders without crashing (enzyme)', () => {
  shallow(<Greeting />);
});

it('renders hello world message (enzyme)', () => {
  const wrapper = shallow(<Greeting />);

  expect(wrapper.type()).toEqual('h1');
  expect(wrapper.children().length).toEqual(1);
  expect(wrapper.childAt(0).node).toEqual('Hello World!');
  
  expect(wrapper.html()).toEqual('<h1>Hello World!</h1>');
});

it('renders hello world snapshot (enzyme)', () => {
  const wrapper = shallow(<Greeting />);
  expect(wrapper).toMatchSnapshot();
});

/// React Test Renderer ///

it('renders hello world snapshot (react-test-renderer)', () => {
  const component = renderer.create(<Greeting />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

