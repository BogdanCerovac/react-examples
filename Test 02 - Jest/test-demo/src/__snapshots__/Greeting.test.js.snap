exports[`test renders hello world snapshot (enzyme) 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <h1>
    Hello World!
</h1>,
  "nodes": Array [
    <h1>
      Hello World!
</h1>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Greeting />,
      "_debugID": 6,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Greeting {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 3,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <h1>
          Hello World!
</h1>,
        "_debugID": 7,
        "_renderedOutput": <h1>
          Hello World!
</h1>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Greeting />,
}
`;

exports[`test renders hello world snapshot (react-test-renderer) 1`] = `
<h1>
  Hello World!
</h1>
`;
